File: demo/src/bag/scripts/record.py
#! /usr/bin/env python
import rospy
import rosbag
from std_msgs.msg import String

if __name__ == "__main__":
    rospy.init_node("recordNode")
    rospy.loginfo("record...")
    bag = rosbag.Bag("./src/bag/hello.bag",'w')

    s = String()
    s.data = "hello world<python>"

    bag.write("chatter",s)
    bag.write("chatter",s)
    bag.write("chatter",s)
    bag.write("chatter",s)
    bag.write("chatter",s)
    bag.write("chatter",s)
    bag.close()

File: src/bag/src/record.cpp
#include "ros/ros.h"
#include "rosbag/bag.h"
#include "std_msgs/String.h"

int main(int argc, char *argv[]) {
    setlocale(LC_ALL,"");

    ros::init(argc,argv,"recordNode");
    ROS_INFO("record...");
    ros::NodeHandle nh;
    rosbag::Bag bag;
    
    bag.open("./src/bag/hello.bag",rosbag::BagMode::Write);
    std_msgs::String msg;
    msg.data = "hello world<C++>";
    
    bag.write("chatter",ros::Time::now(),msg);
    bag.write("chatter",ros::Time::now(),msg);
    bag.write("chatter",ros::Time::now(),msg);
    bag.write("chatter",ros::Time::now(),msg);
    bag.write("chatter",ros::Time::now(),msg);
    bag.write("chatter",ros::Time::now(),msg);
    bag.close();

    return 0;
}
